tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_axis_view.ts

```diff
@@ -0,0 +1,183 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {ChangeDetectionStrategy, Component, Input} from '@angular/core';
+
+import {Dimension, Extent, Scale} from '../lib/public_types';
+import {formatAxisNumber} from './axis_formatter';
+import {
+  getDomX,
+  getDomY,
+  XDimChartView,
+  YDimChartView,
+} from './chart_view_utils';
+
+export abstract class AxisView {
+  trackByTick(index: number, tick: number) {
+    return tick;
+  }
+
+  getTickString(tick: number): string {
+    return formatAxisNumber(tick);
+  }
```

Comment:
The question is a not obvious to me. What is the problem here?

Reply:
Why does the function exist? It only delegates to another function with
the same type signature, and doesn’t really seem to bridge any
abstraction levels (à la [Fowler’s `highlight`][1]). Why not just inline
the implementation at the function’s call sites?

[1]: https://www.martinfowler.com/bliki/FunctionLength.html

Reply:
Ah, it is because Angular templates cannot call a module global function; it can only call a method/static-method attached to the NgComponent class. I think the template gets built to a class of its own. 

Should I name this method `formatAxisNumber` instead?

Reply:
Got it. I think it’s clearer now that it’s actually on an Angular
component. Before, when it was an undocumented function on an
undocumented abstract class without any indication, its purpose was
opaque to me.